libcloud/compute/drivers/oneandone.py

```diff
@@ -0,0 +1,2263 @@
+        'DEPLOYING': NodeState.STARTING,
+    }
+
+    """
+    Core Functions
+    """
+
+    def list_sizes(self):
+        """
+        Lists all sizes
+
+        :return: A list of all configurable node sizes.
+        :rtype: ``list`` of :class:`NodeSize`
+        """
+        sizes = []
+
+        fixed_instances = self._list_fixed_instances()
+        for value in fixed_instances:
+            node_size = self._to_node_size(value)
+            sizes.append(node_size)
+
+        return sizes
+
+    def list_locations(self):
+        """
+        Lists all locations
+
+        :return: ``list`` of :class:`NodeLocation`
+        :rtype: ``list``
+        """
+        datacenters = self.ex_list_datacenters()
+        locations = []
+        for values in datacenters:
+            node_size = self._to_location(values)
+            locations.append(node_size)
+
+        return locations
+
+    def list_images(self, image_type=None):
+        """
+        :return: ``list`` of :class: `NodeImage`
+        :rtype: ``list``
+        """
+        response = self.connection.request(
+            action='server_appliances',
+            method='GET'
+        )
+
+        return self._to_images(response.object, image_type)
+
+    def get_image(self, image_id):
+        response = self.connection.request(
+            action='server_appliances/%s' % image_id,
+            method='GET'
+        )
+        return self._to_image(response.object)
+
+    """
+    Node functions
+    """
+
+    def create_node(self,
+                    name,
+                    image,
+                    ex_fixed_instance_size_id,
+                    location=None,
+                    auth=None,
+                    ex_ip=None,
+                    ex_monitoring_policy_id=None,
+                    ex_firewall_policy_id=None,
+                    ex_loadbalancer_id=None,
+                    ex_description=None,
+                    ex_power_on=None):
+        """
+        Creates a node.
+
+        :param name: The name of the new node
+        :type ```str```
+
+        :param image: Image from list_images
+        :rtype: ``NodeImage``
+
+        :param ex_fixed_instance_size_id:
+        Fixed instance size ID from list_sizes
+        :rtype: ``str``
+
+        :param location: 1&1 Data center Location
+        :type ```NodeLocation```
+
+        :param auth: SSH key or root password
+        :type: :class:`NodeAuthSSHKey` or :class:`NodeAuthPassword`
+
+        :param ex_ip: IP address
+        :type ```str```
+
+        :param ex_ssh_key: SSH Key
+        :type ```str```
+
+        :param password: Password
+        :type ```str```
+
+        :param ex_monitoring_policy_id:
+        :type ```str```
+
+        :param ex_firewall_policy_id:
+        :type ```str```
+
+        :param ex_loadbalancer_id:
+        :type ```str```
+
+        :param ex_description:
+        :type ```str```
+
+        :param ex_power_on:
+        :type ```bool```
+
```

Comment:
most of the docstrings are incorrect, please see http://libcloud.readthedocs.io/en/latest/development.html#docstring-conventions